lucene/core/src/java/org/apache/lucene/store/OutputStreamIndexOutput.java

```diff
@@ -45,23 +46,23 @@
    * @param bufferSize the buffer size in bytes used to buffer writes internally.
    * @throws IllegalArgumentException if the given buffer size is less or equal to <tt>0</tt>
    */
-  public OutputStreamIndexOutput(String resourceDescription, String name, OutputStream out, int bufferSize, boolean useEncryption) {
+  public OutputStreamIndexOutput(String resourceDescription, String name, OutputStream out, int bufferSize, boolean useEncryption, Path path) {
```

Comment:
Is `path` really the best variable name? It's kind a hard to tell what is it used for.

Reply:
Maybe `resourcePath` then?
Anyway, this is `(OutputStream)IndexOutput` a class thing that outputs to some index-related resource.
`name` is the name of the resource we are outputting to.
In analogy, `path` is the path of the resource we are outputting to.

> It's kind a hard to tell what is it used for.

Do you suggest that it's name should reflect the fact it relates to encryption? That's not a good idea I think.
Although I see the design is sort of ugly as you need to provide the `path` if and only if you use the encryption.
(Btw, I could use a regex to parse the path out of the `resourceDescription` but that seemed too hacky.)

Reply:
Yes, I thought it should reflect the relation to the encryption since that's the reason for that variable. But I agree that it's the same case as the `name` variable. Maybe you can replace the `name` variable with the `path` variable and derive the name from it if possible. Otherwise, we can leave it this way.

Reply:
I think I'll move the `initCipher` logic into the child class (`FSIndexOutput` - the only one that uses encryption) and just pass `cipher` to `OutputStreamIndexOutput` constructor, instead of `useencryption + path` pair.